src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -4528,22 +4528,11 @@ private void ParseParameterModifiers(SyntaxListBuilder modifiers, bool isFunctio
 
             var type = this.ParseType();
 
-            var saveTerm = _termState;
-            _termState |= TerminatorState.IsEndOfFieldDeclaration;
-            var variables = _pool.AllocateSeparated<VariableDeclaratorSyntax>();
-            try
-            {
-                this.ParseVariableDeclarators(type, VariableFlags.Fixed, variables, parentKind);
```

Comment:
renamed ParseVariableDeclarators to ParseFieldDeclarationVariableDeclarators  since that's the only case it is used for.  this allowed a lot of simplification.